bundles/org.eclipse.core.runtime/src/org/eclipse/core/runtime/Plugin.java

```diff
@@ -526,6 +527,35 @@ public void setDebugging(boolean value) {
 		}
 	}
 
+	/**
+	 * Try to acquire a single service from the OSGi service factory (e.g.
+	 * DebugOptions, IWorkspace, IPreferenceService, ...)
```

Comment:
Would it be worth adding an @implNote telling it relies on the ServiceTracker, so consumer can get a sense of how they can then inject their own service?

Reply:
I don't think this is specific to how the service is acquired and users can't learn much about if the know how it works internally. "OSGi service factory" actually tells where service are aquired and how to provide own implementations.